tasks/patch_server.rb

```diff
@@ -259,40 +260,51 @@ def reboot_required(family, release)
 # There are no updates available, exit cleanly rebooting if the override flag is set
 if updatecount.zero?
   if reboot_override == true
-    log.info 'Rebooting'
+    log.error 'Rebooting'
     output('Success', reboot, security_only, 'No patches to apply, reboot triggered', '', '', '', pinned_pkgs, starttime)
     $stdout.flush
-    log.info 'No patches to apply, rebooting as requested'
+    log.error 'No patches to apply, rebooting as requested'
     p1 = fork { system('nohup /sbin/shutdown -r +1 2>/dev/null 1>/dev/null &') }
     Process.detach(p1)
   else
     output('Success', reboot, security_only, 'No patches to apply', '', '', '', pinned_pkgs, starttime)
-    log.info 'No patches to apply, exiting'
+    log.error 'No patches to apply, exiting'
   end
   exit(0)
 end
 
 # Run the patching
 if facts['os']['family'] == 'RedHat'
-  log.debug 'Running yum upgrade'
+  log.info 'Running yum upgrade'
   log.debug "Timeout value set to : #{timeout}"
-  yum_output = run_with_timeout("yum #{yum_params} #{securityflag} upgrade -y", timeout, 2)
+  yum_end = ''
+  status, output = run_with_timeout("yum #{yum_params} #{securityflag} upgrade -y", timeout, 2)
+  err(status, 'os_patching/yum', "yum upgrade returned non-zero (#{status}) : #{output}", starttime) if status != 0
 
   if facts['os']['release']['major'].to_i > 5
     # Capture the yum job ID
-    log.debug 'Getting yum job ID'
+    log.info 'Getting yum job ID'
     job = ''
     yum_id, stderr, status = Open3.capture3('yum history')
     err(status, 'os_patching/yum', stderr, starttime) if status != 0
     yum_id.split("\n").each do |line|
-      matchdata = line.to_s.match(/^\s+(\d+)\s/)
+      matchdata = line.to_s.match(/^\s+(\d+)\s*\|\s*[\w\-<> ]*\|\s*([\d:\- ]*)/)
```

Comment:
Could you add a comment about what this is attempting to capture? I don't know what the output here would look like normally anyway.

Reply:
Yep, good call.  Have included more info in a commit that is on the branch now.  Will be in the next release.